documentation/processing.tex

```diff
@@ -1 +1,315 @@
+
+During the work of the processors, they encounter \emph{expression} structures (see \cref{ca_expr:logic}, \cref{mach_expr}). Expressions need to be evaluated for processing to be correctly finished.
+
+There are four specialized instruction processors:
+\paragraph*{Macro IP} looks up for macro definition in HLASM context tables and calls it.
+\paragraph*{Assembler and Machine IP} processes assembler and machine instructions (see \cref{asm_instrs} \cref{mach_instr}) to retain consistency in HLASM context tables.
+
+\paragraph*{Conditional assembly IP} executes conditional assembly instructions (see \cref{ca_instr}). 
+
+\vspace{5mm}
+
+See the current list of processed instruction in \cref{tab06:instr_proc}.
+
+\begin{table}
+	\centering
+	\begin{tabular}{lr}
+		\textbf{IP}                   &                  \textbf{Processed instructions} \\ \toprule
+		\textbf{Assembler}            & *SECT, COM, LOCTR, EQU, DC, DS, COPY, EXTRN, ORG \\
+		\textbf{Machine}              &                                      \emph{NONE} \\
+		\textbf{Macro}                &                                       \emph{ANY} \\
+		\textbf{Conditional Assembly} &    SET*, GBL*, ANOP, ACTR, AGO, AIF, MACRO, MEND \\ \bottomrule
+	\end{tabular}
+	\caption{Table of instructions that are processed by instruction processors.}
+	\label{tab06:instr_proc}
+\end{table}
+
+\subsubsection{CA Expressions}
+\label{ca_expr:logic}
+
+HLASM differentiates two kinds of expressions: \emph{Conditional Assembly} (CA) and \emph{Assembler} (ASM) expressions. CA expressions appear in conditional assembly, which is processed during compilation and has a significant impact on the final generated assembly.
+On the other hand, assembler instructions are the actual instructions that the processor executes.
+
+In this section, we first describe CA expressions following with ASM instructions.
+
+
+
+HLASM evaluates CA expressions during assembly generation. For further details, refer to the \cref{ca_instr}.
+
+We employ the ANTLR 4 Parse-Tree Visitors during the expression evaluation. ANTLR 4 offers two mechanisms for tree-walking: the parse-tree listeners and parse-tree visitors. As the tree-walker encounters a rule, it triggers a \emph{start} function. Similarly, when the walker visits all children of the node, it calls the \emph{finish} function.  The second mechanism is the parse-tree visitor. The visitor lets the programmer control the walk by explicitly calling methods to visit children. We picked the latter approach because we need to have ampler control over the evaluation (such as operator priority).
+
+In this paragraph, we address the representation and functionality of the expressions themselves. Coupling the expressions with grammar and their evaluation in context is further discussed in \cref{ca_expr:eval}.
+
+Each HLASM CA expression has a type. Supported are \emph{arithmetic}, \emph{logic}, \emph{character} expressions. We implement the logic in following classes:
+
```

Comment:
tu by to chcelo nejaky uvodik ktory povie ze expressny su reprezentovane v nejakej (stromovej?) strukture, a ze vobec nejaka evaluation tam nastava